docs/transactions.md

```diff
@@ -105,61 +105,44 @@ This speeds up the application since the thread is free to handle the next trans
-    invoice.Send();
-}
+	// NotSupportedException
+	// ScErrNestedTransactionNotSupported (SCERR4311):
+	// An attempt was made to initialize a new transaction within the scope of an active transaction.
+	// Creating such nested transactions is not supported. Consider reusing the already active transaction or running the new transaction outside the already active transaction's scope.
+	transactor.Transact(db =>
+	{
+	});
+});
 ```
 
-After the `SendInvoice` transaction, you'd expect that the invoice has been committed to the database. That is not the case. To preserve the atomicity of the outer transaction, the changes have to be committed at the same time at the end of the outer transaction's scope. Thus, if the invoice was sent on line 14 and then the whole transaction could roll back in `RemoveFromInventory` and undo `AddInvoiceToDb`. This would cause the customer to receive an invoice that is not stored in the database.
+It is possible to create a new parallel transaction using the following syntax:
 
-Due to this, inner transactions have to specify what to do on commit. To adapt the previous example to specify what to do on commit, we would do this to `SendInvoice`:
+```cs
+using Starcounter.Database.Data;
 
-```csharp
-public void SendInvoice(ITransactor transactor, long productId, Customer customer)
+// Initiating the first transaction.
+transactor.Transact(db1 =>
 {
-    transactor.Transact(db =>
-    {
-        AddInvoiceToDb(invoice);
-    }, 
-    options: new TransactOptions
-    (
-        onCommit: () => invoice.Send())
-    );
-}
-```
-
-With this change, the invoice would be sent whenever the changes in `AddInvoiceToDb` are committed and you can be sure that the invoice would be sent first when the invoice is safely in the database. In this case, it would be when the outer transaction scope terminates. If there was no outer transaction, the invoice would be sent when the transaction with `AddInvoiceToDb` terminates. Thus, `onCommit` ensures that the calls are made in the correct order no matter what.
-
-If you don't know if a transaction will be nested within another transaction, it's always safe to add an empty `onCommit` delegate:
+    var i = db1.Insert<Item>();
 
-```csharp
-var transactor = services.GetRequiredService<ITransactor>();
-
-transactor.Transact(db =>
-{
-    // Read and write to database
-}, new TransactOptions(() => {}));
+    // Initiating the second transaction.
+    // This transaction runs in parallel to the first transaction.
+    // If the first transaction happens to be retried due to a conflict, the second transaction will be executed and committed multiple times.
+    Transaction.Create().Run(db2 =>
+    {
+        // This is an invalid object access.
+        // The object "i" does not exist in the second transaction.
+		// ScErrInvalidObjectAccessAbort (SCERR8006): The transaction was aborted because of an invalid object access.
+        i.Name = "Test";
+    });
+});
```

Comment:
I find this example very confusing, because it mentions transaction snapshot issues, itself a complicated subject, while explaining the complicated subject of nested transactions. I would suggest to not document `Transaction.Create()` at all, and possibly not even expose it in the public API.

The problem is that C# lets developers store things in variables (i.e. database object) that are available in closures in later transaction delegates when the things they represent are in fact not.

Instead we should explain here how to:

1. Reuse the "outer" transaction (by passing the `IDatabaseContext` and remaining in the same synchronization context (necessary and sufficient, right?))
2. Create a new transaction for the "inner" stuff, in a separate part of the code, after the "outer" transaction is finished. This is what developers will most likely have to do. If we want code reuse, we should pass object Oid's and references, not rely on closures (as they in fact do not work).

Hacky developers will find their way around this and schedule transactions in the middle of transactions, but they should not find those kinds of things in our docs.

Reply:
@erikvk please proceed with adjusting the sources, I trust you on this.

Reply:
Yes, that's the plan. When I say "we", I really mean "I" ;)